logrotate.c

```diff
@@ -1210,6 +1225,12 @@ static int copyTruncate(const char *currLog, const char *saveLog, const struct s
     }
 
     if (flags & LOG_FLAG_COPYTRUNCATE) {
+#if defined(WITH_FALLOCATE)
+        if (flags & LOG_FLAG_COPYREDUCE) {
+            message(MESS_NORMAL, "WARNING: copytruncate and copyreduce are mutually exclusive;\n");
+            message(MESS_NORMAL, "copyreduce is disabled.\n");
```

Comment:
I see that the patch was reworked.  Now the diagnostic message is misleading again (and maybe the program logic not fully correct).  It basically tells the user that `copyreduce` was ignored after the flag already affected how `sparse_copy()` was executed.  Please put there also a code comment to explain the `skip_null_head` block in `sparse_copy()`.  Thanks!